### PR TITLE
Various Boltdb/Raft Documentation Updates

### PR DESCRIPTION
**Pull request by mkeeler**
There are 3 commits in this PR. 

1. Adds documentation for the `raft_boltdb.NoFreelistSync` option that was added for 1.11.0-rc.
2. Adds documentation to the telemetry page for all the new boltdb metrics added for 1.11.0-rc.
3. Correct documentation of the consul.raft.fsm.apply metric. Previously we said this was a counter but really its a time measurement.

### CODE REVIEW

File: website/content/docs/agent/options.mdx [hunk 1]
```diff
@@ -1770,6 +1770,15 @@ bind_addr = "{{ GetPrivateInterfaces | include \"network\" \"10.0.0.0/8\" | attr
 - `protocol` ((#protocol)) Equivalent to the [`-protocol` command-line
   flag](#_protocol).
 
+- `raft_boltdb` ((#raft_boltdb)) This is a nested object that allows configuring 
+  options for Rafts BoltDB based log store.
```
Comment (msiege2) [inline website/content/docs/agent/options.mdx:1774]: Raft's

Comment (msiege2) [reply website/content/docs/agent/options.mdx:1774]: and "BoltDB-based".  Two-word adjectives get hyphenated.  One of the few things that stuck with me from HS english.  :)

**Review by msiege2 — APPROVED** [review]
Looks good!

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/522965.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit bcce800f5514cd99b1d4173051ad9f8cfd8606fa onto `release/1.11.x` succeeded!

**Pull request merged**